src/app/component/eco-news/components/create-news/create-news.component.ts

```diff
@@ -118,33 +116,31 @@ export class CreateNewsComponent implements OnInit {
     if ( !filter.isActive ) {
       filter.isActive = true;
       this.isArrayEmpty = false;
-      this.createNewsForm.value.tags.push(filter.name.toLowerCase());
+      this.createNewsForm.value.tags = [...this.createNewsForm.value.tags, filter.name.toLowerCase()];
       this.filtersValidation(filter);
     } else {
       this.removeFilters(filter);
     }
   }
 
   public removeFilters(filter: FilterModel): void {
+    const tagsArray = this.createNewsForm.value.tags;
     if ( filter.isActive ) {
       filter.isActive = false;
-      if (this.createNewsForm.value.tags.length === 1) {
+      if (tagsArray.length === 1) {
         this.isArrayEmpty = true;
       }
```

Comment:
```js
if ( filter.isActive ) {
      filter.isActive = false;
...
```
If `filter.isActive` is `true` it will be `false` according to condition, but it's `false` it will be `false` again. So, this condition doesn't make sense. It all can be rewriten
```js
if (filter.isActive && tagsArray.length === 1) {
  this.isArrayEmpty = true;
};
filter.isActive = false;
```

Reply:
fixed